File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
        unordered_map<int,int>mp, checked;
        
        for(int i : nums)   ++mp[i];
        
        int maxLen = 0;
        
        for(int i = 0; i < nums.size(); i++){
            
            if(checked.find(nums[i]) == checked.end()){
                
                int curLen = 0, start = nums[i];
                
                while(mp.find(start - 1) != mp.end()){
    
                    --start;
                }
                
                while(mp.find(start) != mp.end()){
                    
                    ++curLen;
                    
                    ++start;
                    
                    ++checked[start];
                }
                
                maxLen = max(maxLen, curLen);
                
            }
            
        }
        
        return maxLen;
        
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        
        Map<Integer, Integer> mp = new HashMap<>();
        
        for(int i = 0; i < nums.length; i++){
            
            if(mp.containsKey(nums[i]))     mp.put(nums[i], mp.get(nums[i]) + 1);
            
            else    mp.put(nums[i], 1);            
        }
        
        int res = 0;
        
        Set<Integer> checked = new HashSet<>();
        
        for(Integer key : mp.keySet()){
            
            if(mp.containsKey(key - 1) || checked.contains(key))     continue;
            
            int smallAns = 0, start = key;
            
            while(mp.containsKey(start)){
                
                checked.add(start);
                
                ++smallAns;
                
                ++start;
                
            }
            
            res = Math.max(res, smallAns);
            
        }
        
        return res;
        
    }
}